ext/mysql2/extconf.rb

```diff
@@ -2,7 +2,12 @@
 require 'mkmf'
 
 def asplode lib
-  abort "-----\n#{lib} is missing.  please check your installation of mysql and try again.\n-----"
+  if RUBY_PLATFORM =~ /mingw|mswin/
+    abort "-----\n#{lib} is missing. please check your installation of mysql and try again.\n-----"
+  else
+    abort "-----\n#{lib} is missing. Try 'apt-get install libmysqlclient-dev' or 
+'yum install mysql-devel', check your installation of mysql and try again.\n-----"
+  end
```

Comment:
if we're going this far we may want to include homebrew instructions for osx as well?